index.rst

```diff
@@ -50,7 +50,696 @@
+
+.. _cs_dataunit_joins:
+
+DataUnit Joins
+--------------
+
+The tables in this section represent many-to-many joins between DataUnits
+defined in the previous section that can be generated programmatically.  These
+join tables have no primary key (at least not as part of the common schema),
+and hence cannot be used to label Datasets.
+
++------------------+--------+---------------------------------------------------+
+| *CalibRangeJoin*                                                              |
++==================+========+===================================================+
+| calib_range_id   | uint64 | NOT NULL REFERENCES CalibRange (calib_range_id)   |
++------------------+--------+---------------------------------------------------+
+| visit_id         | uint64 | NOT NULL REFERENCES Visit (visit_id)              |
++------------------+--------+---------------------------------------------------+
+
++--------------------------+--------+-----------------------------------------------------------------+
+| *SensorCalibRangeJoin*                                                                              |
++==========================+========+=================================================================+
+| sensor_calib_range_id    | uint64 | NOT NULL REFERENCES SensorCalibRange (sensor_calib_range_id)    |
++--------------------------+--------+-----------------------------------------------------------------+
+| observed_sensor_id       | uint64 | NOT NULL REFERENCES ObservedSensor (observed_sensor_id)         |
++--------------------------+--------+-----------------------------------------------------------------+
+
+The above two tables define the joins between master calibration Datasets and
+the observations they should be used to calibrate.  These can be defined
+directly as views in on the DataUnit tables:
+
+.. code-block:: sql
+
+    CREATE VIEW CalibRangeJoin AS
+        SELECT
+            Visit.visit_id,
+            CalibRange.calib_range_id
+        FROM
+            Visit INNER JOIN CalibRange ON (
+                (Visit.num BETWEEN CalibRange.first_visit AND CalibRange.last_visit)
+                AND Visit.physical_filter_id = CalibRange.physical_filter_id
+            );
+
+    CREATE VIEW SensorCalibRangeJoin
+        SELECT
+            ObservedSensor.observed_sensor_id,
+            SensorCalibRange.sensor_calib_range_id
+        FROM
+            ObservedSensor INNER JOIN Visit ON (ObservedSensor.visit_id = Visit.visit_id)
+            INNER JOIN SensorCalibRange ON (
+                (Visit.num BETWEEN SensorCalibRange.first_visit AND SensorCalibRange.last_visit)
+                AND Visit.physical_filter_id = SensorCalibRange.physical_filter_id
+            );
+
+The remaining join tables represent the spatial relationships between
+observations and SkyMap entities; records should only be present in these
+tables when the two entities overlap as defined by their ``region`` fields.
```

Comment:
Since a patch is a subset of a tract, why do we need both sensor join tables?

Reply:
I think having both joins clearly makes it easier to use.  It may be less efficient, but that's a non-obvious tradeoff between:
 - storage (and other?) costs of having both
 - having only patch join table, with the additional compute cost of aggregating the patch-join table on the fly to answer tract join queries
 - having only tract join table, with the additional compute cost of rejecting non-overlapping patches that were spuriously rejected by a join against tracts.